android/java/org/chromium/chrome/browser/rewards/BraveRewardsCreatorPanelFragment.java

```diff
@@ -0,0 +1,351 @@
+
+import org.chromium.base.IntentUtils;
+import org.chromium.base.Log;
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.BraveRewardsBalance;
+import org.chromium.chrome.browser.BraveRewardsExternalWallet;
+import org.chromium.chrome.browser.BraveRewardsHelper;
+import org.chromium.chrome.browser.BraveRewardsNativeWorker;
+import org.chromium.chrome.browser.BraveRewardsObserver;
+import org.chromium.chrome.browser.BraveRewardsPublisher;
+import org.chromium.chrome.browser.BraveRewardsPublisher.PublisherStatus;
+import org.chromium.chrome.browser.BraveRewardsSiteBannerActivity;
+import org.chromium.chrome.browser.BraveWalletProvider;
+import org.chromium.chrome.browser.app.BraveActivity;
+import org.chromium.chrome.browser.tab.Tab;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+public class BraveRewardsCreatorPanelFragment extends Fragment
+        implements BraveRewardsHelper.LargeIconReadyCallback, BraveRewardsObserver {
+    private int currentTabId_;
+    private boolean monthlyContribution;
+    private BraveRewardsNativeWorker mBraveRewardsNativeWorker;
+    private final int PUBLISHER_ICON_SIDE_LEN = 50; // dp
+    private BraveRewardsHelper mIconFetcher;
+    private boolean mTippingInProgress; // flag preventing multiple tipping processes
+    private BraveRewardsBannerInfo mBannerInfo;
+
+    private final String TWITTER = "twitter";
+    private final String YOUTUBE = "youtube";
+    private final String TWITCH = "twitch";
+    private final String GITHUB = "github";
+    private final String REDDIT = "reddit";
+    private final String VIMEO = "vimeo";
+
+    public static BraveRewardsCreatorPanelFragment newInstance(
+            int tabId, boolean isMonthlyContribution, String bannerInfo) {
+        BraveRewardsCreatorPanelFragment fragment = new BraveRewardsCreatorPanelFragment();
+        Bundle args = new Bundle();
+        args.putInt(BraveRewardsSiteBannerActivity.TAB_ID_EXTRA, tabId);
+        args.putBoolean(
+                BraveRewardsSiteBannerActivity.IS_MONTHLY_CONTRIBUTION, isMonthlyContribution);
+        args.putString(BraveRewardsSiteBannerActivity.BANNER_INFO_ARGS, bannerInfo);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (getArguments() != null) {
+            currentTabId_ = getArguments().getInt(BraveRewardsSiteBannerActivity.TAB_ID_EXTRA);
+            monthlyContribution = getArguments().getBoolean(
+                    BraveRewardsSiteBannerActivity.IS_MONTHLY_CONTRIBUTION);
+            String bannerInfoJson =
+                    getArguments().getString(BraveRewardsSiteBannerActivity.BANNER_INFO_ARGS);
+            try {
+                mBannerInfo = new BraveRewardsBannerInfo(bannerInfoJson);
+            } catch (JSONException e) {
```

Comment:
assert + log may help with debug

Reply:
@samartnik 
So I will add both assert and log like this right?

} catch (JSONException e) {
assert mBannerInfo == null;
Log.e(TAG, "TippingBanner -> CreatorPanel:onAttach JSONException error " + e);
}